sdk/core/core-lro/src/lroEngine/azureAsyncPolling.ts

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  failureStates,
+  LroResourceLocationConfig,
+  LongRunningOperation,
+  LroBody,
+  LroResponse,
+  LroStatus,
+  RawResponse,
+  successStates
+} from "./models";
+import { isUnexpectedPollingResponse } from "./requestUtils";
+
+function getResponseStatus(rawResponse: RawResponse): string {
+  const { status } = (rawResponse.body as LroBody) ?? {};
+  return status?.toLowerCase() ?? "succeeded";
+}
+
+function isAzureAsyncPollingDone(rawResponse: RawResponse): boolean {
+  const state = getResponseStatus(rawResponse);
+  if (isUnexpectedPollingResponse(rawResponse) || failureStates.includes(state)) {
+    throw new Error(`Operation status: ${state}`);
+  }
+  return successStates.includes(state);
+}
+
+async function sendFinalRequest<TResult>(
+  lro: LongRunningOperation<TResult>,
+  resourceLocation: string,
+  lroResourceLocationConfig?: LroResourceLocationConfig
+): Promise<LroResponse<TResult> | undefined> {
+  switch (lroResourceLocationConfig) {
+    case "original-uri":
+      return lro.sendPollRequest(lro.requestPath, () => true);
+    case "azure-async-operation":
+      return Promise.resolve(undefined);
```

Comment:
this is an async function, so you shouldn't need to wrap `undefined` inside `Promise.resolve`

Reply:
Done.